libcloud/compute/drivers/dimensiondata.py

```diff
@@ -203,12 +243,27 @@ class DimensionDataNodeDriver(NodeDriver):
     DimensionData node driver.
     """
 
+    selected_region = None
     connectionCls = DimensionDataConnection
     name = 'DimensionData'
     website = 'http://www.dimensiondata.com/'
     type = Provider.DIMENSIONDATA
     features = {'create_node': ['password']}
 
+    def __init__(self, key, secret=None, secure=True, host=None, port=None,
+                 region=DEFAULT_REGION, **kwargs):
+
+        if region not in API_ENDPOINTS:
+            raise ValueError('Invalid region: %s' % (region))
+
+        self.selected_region = API_ENDPOINTS[region]
+
+        super(DimensionDataNodeDriver, self).__init__(key=key, secret=secret,
+                                                      secure=secure, host=host,
+                                                      port=port,
+                                                      region=region, **kwargs)
+        self.connection.host = self.selected_region.get('host')
```

Comment:
I believe this won't work correctly when user instantiates multiple drivers with a different region and runs those inside the same process.

You should use `_ex_connection_kwargs` approach. This way, `host` argument will be passe to the Connection class constructor when instantiating it instead of modifying a class level variable. Please see EC2 and some other drivers for an example.